src/helpers/buffers/fsv.rs

```diff
@@ -113,28 +117,54 @@ mod tests {
 
     impl FSBTestExt for FixedSizeByteVec<ELEMENT_SIZE> {
         fn insert_test_data(&mut self, index: usize) {
-            self.insert(index, &test_data_at(index));
+            self.insert("test", index, &test_data_at(index));
         }
     }
 
     #[test]
-    #[should_panic]
     fn insert() {
         let mut v = FixedSizeByteVec::<ELEMENT_SIZE>::new(3);
         v.insert_test_data(0);
-        assert!(v.added(0));
-        v.insert_test_data(2);
-        assert!(v.added(2));
-        assert!(!v.added(1));
+        assert_eq!(v.take(1), Some(test_data_at(0).to_vec()));
+    }
 
+    #[test]
+    fn gap() {
+        let mut v = FixedSizeByteVec::<ELEMENT_SIZE>::new(3);
+        println!("{:?}", v.missing());
+        assert!(v.missing().is_empty());
+        v.insert_test_data(1);
+        assert_eq!(0..1_usize, v.missing());
+    }
+
+    #[test]
+    #[should_panic(expected = "Duplicate send for index 0 on channel \"duplicate\"")]
+    fn duplicate_insert() {
+        let mut v = FixedSizeByteVec::<ELEMENT_SIZE>::new(3);
+        v.insert_test_data(0);
+        v.insert("duplicate", 0, &[10; ELEMENT_SIZE]);
+    }
+
+    #[test]
+    #[should_panic(expected = "Duplicate send for index 0 on channel \"taken\"")]
+    fn insert_taken() {
+        let mut v = FixedSizeByteVec::<ELEMENT_SIZE>::new(3);
+        v.insert_test_data(0);
         assert_eq!(v.take(1), Some(test_data_at(0).to_vec()));
+        v.insert("taken", 0, &[10; ELEMENT_SIZE]);
+    }
 
-        v.insert(1, &[10; ELEMENT_SIZE]);
+    #[test]
+    #[should_panic(expected = "index 10 out of range")]
```

Comment:
does it pass? I can't see where this message is originated from

Reply:
This message comes from the bounds checks on indexing.  It's a tiny bit fragile: we're relying on rust not changing but that seems manageable.